_source/_posts/2021-12-07-electron-javascript.md

```diff
@@ -0,0 +1,298 @@
+
+Next, add two HTML files. Start by adding `index.html`.
+
+```html
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+  <meta charset="UTF-8">
+  <title>A Brief Electron demo with Okta</title>
+</head>
+
+<body>
+<h1>Electron and Okta</h1>
+
+<div class="container">
+  <h4>Please Login</h4>
+
+  <div>
+    <label>Username</label>
+    <input type="text" name="username" id="username"/>
+
+    <label>Password</label>
+    <input type="password" name="password" id="password"/>
+  </div>
+
+  <div>
+    <button id="login-button">Login</button>
+  </div>
+  <div>
+    <label id="error-message"></label>
+  </div>
+
+</div>
+</body>
+
+<style>
+  .container {
+    width: 60%;
+    margin: 0 auto;
+  }
+</style>
+
+<script>
+  document.querySelector("#login-button").addEventListener("click", () => {
+    const username = document.getElementById("username").value;
+    const password = document.getElementById("password").value;
+
+    const data = {
+      username: username,
+      password: password,
+    };
+
+    window.electron.login(data);
+  });
+
+</script>
+
+</html>
+```
+
+The `index.html` page is for users who are not authenticated yet. You are presenting the user with a login screen that collects a username and password and calls `window.electron.login`. As you learned earlier, this function is added in the `preload.js` file.
+
+Finally, add a `home.html` file and provide a landing page for authenticated users.
+
+```html
+<h1>Home</h1>
+
+<div>
+  Congratulations <span id="username">test</span>, You've made it to a secure page using Okta.
+</div>
+
+<div>
+  <button id="logout-button">Logout</button>
+</div>
+
+<script>
+  window.addEventListener('DOMContentLoaded', () => {
+    window.electron.getUser().then((user) => {
+      document.getElementById("username").innerHTML = `${user.profile.firstName} ${user.profile.lastName}`
+    });
+  });
+
+  document.getElementById('logout-button').addEventListener('click', () => {
+    window.electron.logout();
+  });
+</script>
+```
+
+The logout button calls your `electron` API from the `preload.js` file. You are also displaying the user's name from Okta. This helps verify that the user is logged in and who they signed in as.
+
+This page has been left intentionally simple for a very good reason. Electron doesn't care what frontend you use. This could be a full-fledged React application or could have very little JavaScript at all. How you handle managing the application and user state is up to you. But by using the Okta SDKs you can securely authenticate your users, and push them into the application state you need.
```

Comment:
```suggestion
This page has been left intentionally simple for a very good reason. Electron doesn't care what frontend you use. This could be a full-fledged React application or could have very little JavaScript at all. How you handle managing the application and user state is up to you. But, by using the Okta SDKs you can securely authenticate your users, and push them into the application state you need.
```
comma